lib/iris/tests/unit/data_manager/test_DataManager.py

```diff
@@ -452,6 +492,46 @@ def test_nd_lazy_to_nd_lazy(self):
         self.assertTrue(dm.has_lazy_data())
         self.assertArrayEqual(dm.data, lazy_array.compute())
 
+    def test_nd_lazy_to_dataless(self):
+        shape = (2, 3, 4)
+        size = np.prod(shape)
+        real_array = np.arange(size).reshape(shape)
+        lazy_array = as_lazy_data(real_array)
+        dm = DataManager(lazy_array * 10)
+        self.assertTrue(dm.has_lazy_data())
+        dm.data = None
+        self.assertTrue(dm.core_data() is None)
```

Comment:
@ESadek-MO `self.assertIsNone(dm.core_data())` instead?

Repeat this pattern through-out ?